packages/components/src/components/hds/table/index.hbs

```diff
@@ -13,6 +13,9 @@
   <thead class="hds-table__thead">
     {{#if @columns}}
       <Hds::Table::Tr
+        @sortBySelectedItemKey={{@sortBySelectedItemKey}}
+        @sortOrder={{if (eq this.sortBy @sortBySelectedItemKey) this.sortOrder}}
```

Comment:
[note] we may be able to simplify this logic/arguments drilling, but I'll give it another look once the other things has been fixed/simplified so there's less code surface to check.